actions_test.go

```diff
@@ -195,30 +215,61 @@ func TestGetTriggerDefinition(t *testing.T) {
-		"variables":{}
-	}`
-	response := `{"data": {"account": {
-      "customActionsTriggerDefinitions": {
-        "nodes": [
-          {{ template "custom_action_trigger1" }},
-          {{ template "custom_action_trigger2" }}
-        ],
-        {{ template "no_pagination_response" }},
-        "totalCount": 2
-      }
-  }}}`
+	requests := []TestRequest{
+		{`{"query": "query ($after:String!$first:Int!){account{customActionsTriggerDefinitions(after: $after, first: $first){nodes{action{aliases,id},aliases,description,filter{id,name},id,manualInputsDefinition,name,owner{alias,id},published,timestamps{createdAt,updatedAt},accessControl,responseTemplate},pageInfo{hasNextPage,hasPreviousPage,startCursor,endCursor},totalCount}}}",
+			{{ template "pagination_initial_query_variables" }}
+			}`,
+			`{
+				"data": {
+					"account": {
+						"customActionsTriggerDefinitions": {
+							"nodes": [
+								{
+									{{ template "custom_action_trigger1_response" }}
+								},
+								{
+									{{ template "custom_action_trigger2_response" }} 
+								}
+							],
+							{{ template "pagination_initial_pageInfo_response" }},
+							"totalCount": 2
+						  }}}}`},
+		{`{"query": "query ($after:String!$first:Int!){account{customActionsTriggerDefinitions(after: $after, first: $first){nodes{action{aliases,id},aliases,description,filter{id,name},id,manualInputsDefinition,name,owner{alias,id},published,timestamps{createdAt,updatedAt},accessControl,responseTemplate},pageInfo{hasNextPage,hasPreviousPage,startCursor,endCursor},totalCount}}}",
+			{{ template "pagination_second_query_variables" }}
+			}`,
+			`{
+				"data": {
+					"account": {
+						"customActionsTriggerDefinitions": {
+							"nodes": [
+								{
+									{{ template "custom_action_trigger3_response" }}
+								}
+							],
+							{{ template "pagination_second_pageInfo_response" }},
+							"totalCount": 1
+						  }}}}`},
+	}
+
+	// An easy way to see the results of templating is by uncommenting this
+	//fmt.Println(Templated(requests[0].Response))
+	//panic(true)
+
+	//"{account{customActionsTriggerDefinitions(after: $after, first: $first){nodes{action{aliases,id},aliases,description,filter{id,name},id,manualInputsDefinition,name,owner{alias,id},published,timestamps{createdAt,updatedAt},accessControl,responseTemplate}}}}",
 
-	client := ABetterTestClient(t, "custom_actions/list_triggers", request, response)
+	client := APaginatedTestClient(t, "custom_actions/list_triggers", requests...)
 	// Act
 	triggers, err := client.ListTriggerDefinitions(nil)
+	result := triggers.Nodes
 	// Assert
 	autopilot.Ok(t, err)
-	autopilot.Equals(t, 2, triggers.TotalCount)
-	autopilot.Equals(t, "Release", triggers.Nodes[1].Name)
-	autopilot.Equals(t, "Uses Ruby", triggers.Nodes[1].Filter.Name)
-	autopilot.Equals(t, "123456789", string(triggers.Nodes[1].Owner.Id))
+	autopilot.Equals(t, 3, len(result))
```

Comment:
This actually brings to light we are not handling `.TotalCount` correctly anymore.  We should have a sync on this and discuss

Reply:
Thanks for the input, here is the refactor for using TotalCount!